lib/asbind-instance/asbind-instance.js

```diff
@@ -26,7 +24,32 @@ const traverseObjectAndRunCallbackForFunctions = (
       );
     }
   });
-};
+}
+
+async function compileStreaming(source) {
+  if (WebAssembly.compileStreaming) {
+    return WebAssembly.compileStreaming(source);
+  }
+  source = await Promise.resolve(souce);
+  if (source instanceof Response) {
+    source = await source.arrayBuffer();
+  }
+  return WebAssembly.compile(source);
+}
+
+function extractTypeDescriptor(module) {
+  const sections = WebAssembly.Module.customSections(module, SECTION_NAME);
```

Comment:
Woah, I didn't know you could grab custom sections from JS :joy:

Reply:
Neither did I!! All credit to @dcodeio here